src/Workspaces/CSharp/Portable/Extensions/BasePropertyDeclarationSyntaxExtensions.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Microsoft.CodeAnalysis.CSharp.Extensions
+{
+    internal static class BasePropertyDeclarationSyntaxExtensions
+    {
+        public static BasePropertyDeclarationSyntax WithAccessorList(this BasePropertyDeclarationSyntax node, AccessorListSyntax accessorList)
+        {
+            if (node != null)
+            {
+                switch (node.Kind())
+                {
+                    case SyntaxKind.PropertyDeclaration: return ((PropertyDeclarationSyntax)node).WithAccessorList(accessorList);
+                    case SyntaxKind.IndexerDeclaration: return ((IndexerDeclarationSyntax)node).WithAccessorList(accessorList);
+                    case SyntaxKind.EventDeclaration: return ((EventDeclarationSyntax)node).WithAccessorList(accessorList);
+                }
+            }
+
+            return node;
+        }
+
+        /// <summary>
+        /// Available if <paramref name="node"/> is <see cref="PropertyDeclarationSyntax"/> or <see cref="IndexerDeclarationSyntax"/>.
+        /// </summary>
+        public static SyntaxToken TryGetSemicolonToken(this BasePropertyDeclarationSyntax node)
```

Comment:
I'm not exactly sure this is the right solution. SemicolonToken is on properties & indexers but not event declarations, so I'm not sure if it belongs here. TryGetSemicolonToken is OK-ish but TryWithSemicolonToken seems ugly. But it's still useful to help avoid code duplication.

Reply:
Seems fine.  It's an internal helper.